pkg/apis/submariner.io/v1/types.go

```diff
@@ -271,3 +272,64 @@ type ClusterGlobalEgressIPList struct {
 
 	Items []ClusterGlobalEgressIP `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GlobalIngressIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of desired behavior of GlobalIngressIP object.
+	Spec GlobalIngressIPSpec `json:"spec"`
+
+	// Observed status of GlobalIngressIP. Its a read-only field.
+	// +optional
+	Status GlobalIngressIPStatus `json:"status,omitempty"`
+}
+
+type GlobalIngressIPSpec struct {
+	// TargetType can be ClusterIPService or a HeadlessServicePod
+	Target TargetType `json:"target"`
+
+	// If the TargetType points to a Pod that belongs to a Headless Service, it uses the
+	// name of the corresponding Service.
+	// +Optional
+	ServiceName string `json:"serviceName,omitempty"`
```

Comment:
As discussed during the meeting, lets use the ServiceName as LocalObjectReference and store the ClusterIPServiceName or HeadlessServiceName all the time. The TargetRef will be an optional field and will be populated only for HeadlessServicePods.

Reply:
Type will be a pointer *corev1.LocalObjectReference `json:"objRef,omitempty"